src/ccmain/ltrresultiterator.cpp

```diff
@@ -124,19 +121,20 @@ float LTRResultIterator::Confidence(PageIteratorLevel level) const {
     case RIL_TEXTLINE:
       do {
         best_choice = res_it.word()->best_choice;
-        ASSERT_HOST(best_choice != nullptr);
         mean_certainty += best_choice->certainty();
         ++certainty_count;
         res_it.forward();
       } while (res_it.row() == res_it.prev_row());
       break;
     case RIL_WORD:
-      mean_certainty += best_choice->certainty();
-      ++certainty_count;
+      best_choice = res_it.word()->best_choice;
+      mean_certainty = best_choice->certainty();
+      certainty_count = 1;
```

Comment:
Doesn't change behavor?

`++certainty_count;` vs `certainty_count = 1;`

Reply:
Initial value 0 and a single increment gives value 1. There is no loop around it.